dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/LimitHelper.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2024 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.dockstore.webservice.helpers;
+
+import io.dockstore.webservice.CustomWebApplicationException;
+import io.dockstore.webservice.core.Version;
+import org.apache.http.HttpStatus;
+
+public final class LimitHelper {
+
+    public static final long BYTES_PER_MEGABYTE = 1_000_000L;
+    public static final long MAXIMUM_VERSION_FILE_SIZE = 10 * BYTES_PER_MEGABYTE;
+
+    private LimitHelper() {
+        // This space intentionally left blank.
+    }
+
+    public static void checkVersion(Version<?> version) {
+        if (totalFileSize(version) > MAXIMUM_VERSION_FILE_SIZE) {
+            String message = "A version must contain less than %.1fMB of files".formatted(MAXIMUM_VERSION_FILE_SIZE / (double)BYTES_PER_MEGABYTE);
+            throw new CustomWebApplicationException(message, HttpStatus.SC_BAD_REQUEST);
+        }
+    }
+
+    private static long totalFileSize(Version<?> version) {
+        return version.getSourceFiles().stream().mapToLong(file -> file.getContent().length()).sum();
```

Comment:
There are over 1K source files with `null` content in a recent DB dump. Can they cause an NPE here?

Reply:
Why yes, they can!  Good catch, Charles!

Reply:
Related: I am considering making the "content" field of SourceFile not nullable.  Not sure it ever makes sense for a Sourcefile's content to be null.  Thoughts?

Reply:
> Related: I am considering making the "content" field of SourceFile not nullable. Not sure it ever makes sense for a Sourcefile's content to be null. Thoughts?

Do we have `SourceFile`s corresponding to not-on-GitHub files? It seems like we shouldn't, but if we do have that, it seems like the content should be:

- `null` if the file isn't on GitHub
- empty if the file is on GitHub, but no content
- the content if the file is on GitHub with content.

I guess we need to figure out how the `null` content got there.

Reply:
> I guess we need to figure out how the `null` content got there.

At least some happened when the file content retrieval code returned `null`, which happened for large files.  Quote from https://github.com/dockstore/dockstore/pull/5893:

> the GitHub API query used to retrieve a file's contents returns null when the file's size is >= 1MB. This null is stored as the corresponding SourceFile's content.

This was a bug, and has been fixed.  Possibly, there are other causes, our file retrieval code doesn't feel super robust...

I will address this in an upcoming PR.